### PR TITLE
CRM-19303 : CKEditor configuration can't be edited on a Drupal multisite installation

### PR DESCRIPTION
**Pull request by MiltoxBeyond**
Single Merge

Instead of assuming the default folder exists, it checks, if not it will search nearby directories for the files.

---
- [CRM-19303: CKEditor configuration can't be edited on a Drupal multisite installation](https://issues.civicrm.org/jira/browse/CRM-19303)

### CODE REVIEW

**Comment by colemanw** [comment]
@civicrm-builder add to whitelist

**Review by colemanw — CHANGES_REQUESTED** [review]
Code looks good to me. I haven't tested the functionality.

**Review by colemanw — APPROVED** [review]

**Review by colemanw — CHANGES_REQUESTED** [review]
Actually this can't be merged until the [code style problems](https://test.civicrm.org/job/CiviCRM-Core-PR/11819/) are resolved.

**Comment by MiltoxBeyond** [comment]
Style changes committed.  Fixed all warnings.

**Comment by colemanw** [comment]
@civicrm-builder test this please

**Comment by colemanw** [comment]
@civicrm-builder test this please

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
@MiltoxBeyond we are getting an error RuntimeException: Cannot resolve path using "civicrm.files.url" in Civi\Core\Paths->getVariable() (line 83 of /home/jenkins/buildkit/build/core-9088-7x4bp/sites/all/modules/civicrm/Civi/Core/Paths.php. I suspect this is related to your PR

**Comment by MiltoxBeyond** [comment]
Do you have a stack trace for that error so I can track it down?  I only
added an extra check when the path doesn't exist otherwise it returns the
same values.

**Comment by monishdeb** [comment]
@MiltoxBeyond here's the stacktrace 
```
WD menu: RuntimeException: Cannot resolve path using                     [error]
"civicrm.files.url" in Civi\Core\Paths->getVariable() (line 83 of
/home/jenkins/buildkit/build/core-9088-7x4bp/sites/all/modules/civicrm/Civi/Core/Paths.php).
WD php: RuntimeException: Cannot resolve path using                      [error]
"civicrm.files.url" in Civi\Core\Paths->getVariable() (line 83 of
/home/jenkins/buildkit/build/core-9088-7x4bp/sites/all/modules/civicrm/Civi/Core/Paths.php).
Cannot modify header information - headers already sent by (output     [warning]
started at
/home/jenkins/buildkit/vendor/drush/drush/includes/output.inc:38)
bootstrap.inc:1471
RuntimeException: Cannot resolve path using "civicrm.files.url" in Civi\Core\Paths->getVariable() (line 83 of /home/jenkins/buildkit/build/core-9088-7x4bp/sites/all/modules/civicrm/Civi/Core/Paths.php).
Drush command terminated abnormally due to an unrecoverable error.       [error]
```

For detail see related test build console https://test.civicrm.org/job/CiviCRM-Core-PR/11931/console

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/Utils/System/DrupalBase.php [hunk 1]
```diff
@@ -568,4 +568,90 @@ public function parseDrupalSiteName($civicrm_root) {
...
+          continue;
+        }
+
+        //Check if files path exists...
+        if ($this->checkFilesExists($basepath, $folder)) {
+          $correct = $folder;
+          break;
+        }
+      }
+    }
```
Comment (seamuslee001) [inline CRM/Utils/System/DrupalBase.php]: @MiltoxBeyond you need to return something from this function if $this->filesUrl is not set 

so what i have tested and seems to work is adding in at the end here 

```php
  $this->filesUrl = self::getUrl($basepath, $correct);
  return $this->filesUrl;
```

Comment (colemanw) [inline CRM/Utils/System/DrupalBase.php]: @MiltoxBeyond looks like the above is holding up this PR.

**Comment by eileenmcnaughton** [comment]
closing as replaced by #9409

**Pull request closed without merging**